src/Symfony/Component/HttpKernel/Controller/ArgumentResolver/ServiceValueResolver.php

```diff
@@ -39,6 +39,12 @@ public function supports(Request $request, ArgumentMetadata $argument)
 
         if (\is_array($controller) && \is_callable($controller, true) && \is_string($controller[0])) {
             $controller = $controller[0].'::'.$controller[1];
+        } elseif (!\is_string($controller) || '' === $controller) {
+            return false;
+        }
+
+        if ('\\' === $controller[0]) {
+            $controller = ltrim($controller, '\\');
         }
 
         return \is_string($controller) && $this->container->has($controller) && $this->container->get($controller)->has($argument->getName());
```

Comment:
`is_string($controller)` is useless here now, as it will always be `true` (due to the new check added above)

Reply:
right, thx!